docs/user_guide/confidential_computing/cc_architecture.rst

```diff
@@ -0,0 +1,1063 @@
+
+CoCo Trustee Architecture
+=========================
+
+Components
+----------
+
+- **Key Broker Service**: The KBS is a server that facilitates remote attestation and secret delivery. Its role is similar to that of the Relying Party in the RATS model.
+- **Attestation Service**: The AS verifies TEE evidence. In the RATS model, this is a Verifier.
+- **Reference Value Provider Service**: The RVPS manages reference values used to verify TEE evidence. This is related to the discussion in section 7.5 of the RATS document.
+- **KBS Client Tool**: This is a simple tool which can be used to test or configure the KBS and AS.
+
+Note: We are not using the RVPS component. There are no supported APIs to use. We are not using the CDH (Confidential Data Hub) component.
+
+CoCo Trustee Services
+---------------------
+
+- Create reference values
+- Login credentials
+- Role-based Access Control (RBAC) ⇒ missing
+- Identity namespace ⇒ use “path” for now ⇒ missing proper identity namespace
+- Retrieval reference value
+- Identity namespace ⇒ use “path” for now ⇒ missing proper identity namespace
+- Access control ⇒ missing
+- TLS communication (PR merged) ⇒ fixed
+
+Trustee Policies
+================
+
+The "trustee policy" refers to the rules and configurations governing how secrets are released and how the trustworthiness of a confidential workload is verified before granting access to sensitive data. It involves two main types of policies: resource policies and attestation policies.
+
+- **Resource Policies**: These policies determine which secrets are released to a specific workload, typically scoped to the container. They control what secrets are available to the workload, ensuring that only necessary information is provided.
+- **Attestation Policies**: These policies define how the claims about the Trusted Computing Base (TCB) are compared to reference values to determine the trustworthiness of the workload. They specify how the attestation process verifies that the workload is running in a trusted environment.
+
+What we do: Currently, we only need to use resource policy, we will use the default attestation policy.
+
+One can set the policy to the needed measurement (hash values) or referring to the reference values. We choose to use the resource policy for now.
+
+Set Policy
+----------
+
+Here is a policy example. The resource policy we set to ensure only CVM with the measurement matching the value can get the resource (the key for LUKS).
+
+.. code-block:: text
+
+   package policy
+   default allow = false
+   allow {
+       input["submods"]["cpu"]["ear.veraison.annotated-evidence"]["snp"]["measurement"] == "Cwa8qBJimP2freTTrrpvAZVbEQEyAhPY4fZGgSn9z4qtt0CAGmcS+Otz96qQZ92k"
+   }
+
+And the command to set this policy into the Trustee service.
+
+.. code-block:: bash
+
+   #!/usr/bin/env bash
+
+   TRUSTEE_ADDRESS=trustee-azsnptpm.eastus.cloudapp.azure.com
+   PORT=8999
+
```

Comment:
This Trustee address is our own development trustee.  In customer's own deployment, this trustee should change to their self-hosting trustee instance.  Suggest to add a few lines mentioning this difference.